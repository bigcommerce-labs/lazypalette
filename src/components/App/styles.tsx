import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
