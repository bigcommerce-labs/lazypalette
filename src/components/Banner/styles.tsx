import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
