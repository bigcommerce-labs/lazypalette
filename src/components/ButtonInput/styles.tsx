import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
