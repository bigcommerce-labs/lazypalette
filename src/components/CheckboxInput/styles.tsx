import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
