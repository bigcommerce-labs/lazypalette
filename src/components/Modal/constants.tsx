export const enum Messages {
    Reset = 'This theme has unpublished changes. Do you want to proceed?',
    Variation = 'This variation has unpublished changes. Do you want to proceed?',
}
