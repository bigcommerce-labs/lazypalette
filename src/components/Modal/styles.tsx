import styled from 'styled-components';

interface OverlayProps {
    isTransparent?: boolean;
}

export const Close = styled.div`
    cursor: pointer;
    text-align: center;
    width: 2rem;
    height: 2rem;
    margin-right: .25rem;
    display: flex;
    justify-content: center;
    align-items: center;

    :hover {
      cursor: pointer;
      background: ${({ theme }) => theme.colors.darkHover};
      border-radius: 50%;
    }
`;

Close.defaultProps = {
    theme: {
        colors: {
            darkHover: '#D3D9E5',
            primaryText: '#AAAAAA',
        },
    },
};

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow-y: auto;
    z-index: ${({ theme }) => theme.layers.higher};
`;

Container.defaultProps = {
    theme: {
        layers: {
            higher: 100,
        },
    },
};

export const Overlay = styled.div.attrs<OverlayProps>({})`
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    overflow-y: auto;
    opacity: ${({ isTransparent }) => isTransparent ? 0 : 0.45};
    background: #000000;
    z-index: ${({ theme }) => theme.layers.higher};
`;

Overlay.defaultProps = {
    theme: {
        layers: {
            higher: 100,
        },
    },
};

export const Content = styled.div`
    color: ${({ theme }) => theme.colors.primaryText};
    line-height: 1.5rem;
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.normal};
    margin: 0;
    padding: 0;
`;

Content.defaultProps = {
    theme: {
        colors: {
            primaryText: '#AAAAAA',
        },
        typography: {
            fontSize: {
                small: '16px',
            },
            fontWeight: {
                normal: 400,
            },
        },
    },
};

export const Header = styled.div`
    align-items: stretch;
    display: flex;
    flex-shrink: 0;
    justify-content: space-between;
    line-height: 2rem;
    margin: 0 -0.75rem 1rem 0;
    padding: 0;
`;

export const Title = styled.h3`
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.larger};
    font-weight: ${({ theme }) => theme.typography.fontWeight.normal};
    margin: 0;
`;

Title.defaultProps = {
    theme: {
        colors: {
            primaryText: '#303540',
        },
        typography: {
            fontSize: {
                larger: '24px',
            },
            fontWeight: {
                normal: 600,
            },
        },
    },
};

export const ModalBox = styled.div`
    background-color: ${({ theme }) => theme.colors.empty};
    border: 1px solid ${({ theme }) => theme.colors.guideText};
    border-radius: 4px;
    box-shadow: ${({ theme }) => theme.elevation.floating};
    box-sizing: border-box;
    overflow: auto;
    padding: 1.5rem;
    position: relative;
    width: 30rem;
    z-index: ${({ theme }) => theme.layers.highest};
`;

ModalBox.defaultProps = {
    theme: {
        colors: {
            empty: '#FFFFFF',
            guideText: '#ABB1BE',
        },
        elevation: {
            floating: '0 2px 12px rgba(48, 53, 64, 0.2)',
        },
        layers: {
            highest: 500,
        },
    },
};

export const ModalBody = styled.div`
`;

export const ConfirmButtons = styled.div`
    display: inline-flex;
    justify-content: flex-end;
`;

export const ConfirmButton = styled.div`
    margin-left: 1rem;
`;

export const ModalView = styled.div`
    max-width: 35rem;
`;

export const ModalFooter = styled.div`
    display: flex;
    justify-content: flex-end;
    margin: 1.5rem 0 0 0;
`;
