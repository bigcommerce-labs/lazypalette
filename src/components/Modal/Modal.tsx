import React, { MouseEvent, MouseEventHandler, PureComponent } from 'react';

import { Close, Container, Content, Header, ModalBox, NavItem, Overlay, Title } from './styles';

interface ModalProps extends Partial<{
    backLink: string;
    children: any;
    isTransparent: boolean;
    title: string;
    onClose: MouseEventHandler<HTMLDivElement>;
    overlayClose: MouseEventHandler<HTMLDivElement>;
}> {}

class Modal extends PureComponent<ModalProps> {
    static defaultProps: ModalProps = {
        isTransparent: true,
    };

    handleContainerClick = (event: MouseEvent<HTMLDivElement>) => {
        if (this.props.overlayClose) {
            this.props.overlayClose(event);
        }
    };

    handleCloseClick = (event: MouseEvent<HTMLDivElement>) => {
        if (this.props.onClose) {
            this.props.onClose(event);
        }
    };

    handleModalClick = (event: MouseEvent<HTMLDivElement>) => {
        event.stopPropagation();
    };

    render() {
        const {
            backLink,
            children,
            isTransparent,
            title,
        } = this.props;

        return (
            <Container onClick={this.handleContainerClick}>
                <Overlay isTransparent={isTransparent} />
                <ModalBox onClick={this.handleModalClick}>
                    {title &&
                        <Header>
                            <Title>{title}</Title>
                            <Close onClick={this.handleCloseClick}/>
                        </Header>
                    }
                    {backLink &&
                        <NavItem to={backLink} replace />
                    }
                    <Content>{children}</Content>
                </ModalBox>
            </Container>
        );
    }
}

export default Modal;
