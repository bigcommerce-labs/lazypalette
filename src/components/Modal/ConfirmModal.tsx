import React, { PureComponent } from 'react';

import ButtonInput from '../ButtonInput/ButtonInput';

import { Confirm, ConfirmBody, ConfirmButtons, ConfirmFooter } from './styles';
import Modal from './Modal';

interface ConfirmModalProps extends Partial<{
    body: string;
    title: string;
    primaryAction(): void;
    secondaryAction(): void;
    overlayClose(): void;
}> {}

const CTAButtons = ({ primaryAction, secondaryAction }: ConfirmModalProps) => (
    <ConfirmButtons>
        <ButtonInput
            onClick={primaryAction}
            classType="primary"
            type="button"
        >
            Cancel
        </ButtonInput>
        <ButtonInput
            onClick={secondaryAction}
            type="button"
        >
            OK
        </ButtonInput>
    </ConfirmButtons>
);

class ConfirmModal extends PureComponent<ConfirmModalProps> {
    static defaultProps: ConfirmModalProps = {
        title: 'Please Confirm:',
    };

    render() {
        const {
            body,
            primaryAction,
            secondaryAction,
            overlayClose,
            title,
        } = this.props;

        return (
            <Modal
                onClose={primaryAction}
                overlayClose={overlayClose}
                title={title}
                isTransparent={false}
            >
                <Confirm>
                    {body &&
                        <ConfirmBody>{body}</ConfirmBody>}
                    <ConfirmFooter>
                        <CTAButtons
                            primaryAction={primaryAction}
                            secondaryAction={secondaryAction}
                        />
                    </ConfirmFooter>
                </Confirm>
            </Modal>
        );
    }
}

export default ConfirmModal;
