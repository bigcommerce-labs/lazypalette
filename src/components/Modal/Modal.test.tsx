import { mount, shallow } from 'enzyme';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import { Container, Header, ModalBox, NavItem } from './styles';
import Modal from './Modal';

describe('Modal', () => {
    const theme = {
        colors: {
            background: '#f0f',
            empty: '#FFFFFF',
        },
        elevation: {
            raised: '0 1px 6px rgba(48, 53, 64, 0.2)',
        },
        layers: {
            high: '50',
        },
        typography: {
            fontWeight: {
                bold: '600',
            },
        },
    };

    it('renders', () => {
        const mockHandler = jest.fn();

        const TestChildren = () => (
            <div>some inner content</div>
        );

        const modal = shallow(
            <Modal
                backLink=""
                isTransparent={false}
                title="Test Modal"
                onClose={mockHandler}
            >
                <TestChildren />
            </Modal>
        );

        expect(modal).toMatchSnapshot();
    });

    describe('Container', () => {
        describe('onClose', () => {
            it('should clear the uiWindow when Container clicked', () => {
                const mockHandler = jest.fn();

                const TestChildren = () => (
                    <div>some inner content</div>
                );

                const modal = shallow(
                    <Modal
                        backLink=""
                        isTransparent={false}
                        title="Test Modal"
                        overlayClose={mockHandler}
                    >
                        <TestChildren />
                    </Modal>
                );

                const container = modal.find(Container);
                const event = {
                    target: {
                        value: 'testing',
                    },
                };

                container.simulate('click', event);
                expect(mockHandler).toHaveBeenCalledTimes(1);
            });

            it('should not trigger when modalBox clicked', () => {
                const mockHandler = jest.fn();

                const TestChildren = () => (
                    <div>some inner content</div>
                );

                const modal = mount(
                    <ThemeProvider theme={theme}>
                        <Modal
                            backLink=""
                            isTransparent={false}
                            title="Test Modal"
                            onClose={mockHandler}
                        >
                            <TestChildren />
                        </Modal>
                    </ThemeProvider>
                );

                const modalBox = modal.find(ModalBox);
                const event = {
                    target: {
                        value: 'testing',
                    },
                };

                modalBox.simulate('click', event);
                expect(mockHandler).not.toHaveBeenCalled();
            });
        });
    });

    describe('title prop', () => {
        const mockHandler = jest.fn();
        const TestChildren = () => (
            <div>some inner content</div>
        );

        it('should render Header styled-component when prop set', () => {
            const title = 'Test Modal';
            const modal = shallow(
                <Modal
                    title={title}
                    onClose={mockHandler}
                >
                    <TestChildren />
                </Modal>
            );

            expect(modal.find(Header).prop('children')).toEqual(title);
        });

        it('should not render Header when not set', () => {
            const title = undefined;

            const modal = shallow(
                <Modal
                    title={title}
                    onClose={mockHandler}
                >
                    <TestChildren />
                </Modal>
            );

            expect(modal.find(Header).exists()).toEqual(false);
        });
    });

    describe('backLink prop', () => {
        const mockHandler = jest.fn();
        const TestChildren = () => (
            <div>some inner content</div>
        );

        it('should render NavItem styled-component when prop set', () => {
            const backLink = '/design';
            const modal = shallow(
                <Modal
                    backLink={backLink}
                    onClose={mockHandler}
                >
                    <TestChildren />
                </Modal>
            );

            expect(modal.find(NavItem).prop('to')).toEqual(backLink);
        });

        it('should not render NavItem when not set', () => {
            const backLink = undefined;
            const modal = shallow(
                <Modal
                    backLink={backLink}
                    onClose={mockHandler}
                >
                    <TestChildren />
                </Modal>
            );

            expect(modal.find(NavItem).exists()).toEqual(false);
        });
    });
});
