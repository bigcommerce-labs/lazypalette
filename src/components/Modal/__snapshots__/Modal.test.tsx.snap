// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<styled.div
  onClick={[Function]}
  theme={
    Object {
      "layers": Object {
        "higher": 100,
      },
    }
  }
>
  <styled.div
    isTransparent={false}
    theme={
      Object {
        "layers": Object {
          "higher": 100,
        },
      }
    }
  />
  <styled.div
    onClick={[Function]}
    theme={
      Object {
        "colors": Object {
          "empty": "#FFFFFF",
          "guideText": "#ABB1BE",
        },
        "elevation": Object {
          "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
        },
        "layers": Object {
          "highest": 500,
        },
      }
    }
  >
    <styled.div>
      <styled.h3
        theme={
          Object {
            "colors": Object {
              "primaryText": "#303540",
            },
            "typography": Object {
              "fontSize": Object {
                "larger": "24px",
              },
              "fontWeight": Object {
                "normal": 600,
              },
            },
          }
        }
      >
        Test Modal
      </styled.h3>
      <styled.div
        onClick={[Function]}
        theme={
          Object {
            "colors": Object {
              "darkHover": "#D3D9E5",
              "primaryText": "#AAAAAA",
            },
          }
        }
      >
        <Icon
          glyph="closeX"
          primaryColor="#303540"
          size="small"
        />
      </styled.div>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "primaryText": "#AAAAAA",
          },
          "typography": Object {
            "fontSize": Object {
              "small": "16px",
            },
            "fontWeight": Object {
              "normal": 400,
            },
          },
        }
      }
    >
      <TestChildren />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Modal title prop should not render Header without close button when onClose is not set 1`] = `
<styled.div
  onClick={[Function]}
  theme={
    Object {
      "layers": Object {
        "higher": 100,
      },
    }
  }
>
  <styled.div
    isTransparent={true}
    theme={
      Object {
        "layers": Object {
          "higher": 100,
        },
      }
    }
  />
  <styled.div
    onClick={[Function]}
    theme={
      Object {
        "colors": Object {
          "empty": "#FFFFFF",
          "guideText": "#ABB1BE",
        },
        "elevation": Object {
          "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
        },
        "layers": Object {
          "highest": 500,
        },
      }
    }
  >
    <styled.div>
      <styled.h3
        theme={
          Object {
            "colors": Object {
              "primaryText": "#303540",
            },
            "typography": Object {
              "fontSize": Object {
                "larger": "24px",
              },
              "fontWeight": Object {
                "normal": 600,
              },
            },
          }
        }
      />
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "primaryText": "#AAAAAA",
          },
          "typography": Object {
            "fontSize": Object {
              "small": "16px",
            },
            "fontWeight": Object {
              "normal": 400,
            },
          },
        }
      }
    >
      <TestChildren />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Modal title prop should render Header with close button when onClose is set 1`] = `
<styled.div
  onClick={[Function]}
  theme={
    Object {
      "layers": Object {
        "higher": 100,
      },
    }
  }
>
  <styled.div
    isTransparent={true}
    theme={
      Object {
        "layers": Object {
          "higher": 100,
        },
      }
    }
  />
  <styled.div
    onClick={[Function]}
    theme={
      Object {
        "colors": Object {
          "empty": "#FFFFFF",
          "guideText": "#ABB1BE",
        },
        "elevation": Object {
          "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
        },
        "layers": Object {
          "highest": 500,
        },
      }
    }
  >
    <styled.div>
      <styled.h3
        theme={
          Object {
            "colors": Object {
              "primaryText": "#303540",
            },
            "typography": Object {
              "fontSize": Object {
                "larger": "24px",
              },
              "fontWeight": Object {
                "normal": 600,
              },
            },
          }
        }
      >
        Test Modal
      </styled.h3>
      <styled.div
        onClick={[Function]}
        theme={
          Object {
            "colors": Object {
              "darkHover": "#D3D9E5",
              "primaryText": "#AAAAAA",
            },
          }
        }
      >
        <Icon
          glyph="closeX"
          primaryColor="#303540"
          size="small"
        />
      </styled.div>
    </styled.div>
    <styled.div
      theme={
        Object {
          "colors": Object {
            "primaryText": "#AAAAAA",
          },
          "typography": Object {
            "fontSize": Object {
              "small": "16px",
            },
            "fontWeight": Object {
              "normal": 400,
            },
          },
        }
      }
    >
      <TestChildren />
    </styled.div>
  </styled.div>
</styled.div>
`;
